test/functional/inspec_json_profile_test.rb

```diff
@@ -178,4 +178,18 @@
       assert_exit_code 0, out
     end
   end
+
+  # regression #4615
+  describe "inspec json with an archived profile containing an empty file" do
+    it "does not crash" do
+      out = inspec("json " + File.join(profile_path, "contains-empty-file-0.1.0.tar.gz"))
+
+      assert_equal "contains-empty-file", JSON.load(out.stdout)["name"]
+
+      _(out.stderr).must_be_empty ""
+
+      assert_exit_code 0, out
+    end
```

Comment:
I think this is overkill. We don't need a whole integration test to verify that we handle tarballs with empty files correctly. I'm going to work out an alternative because this should be a 1 line impl change, and very short (actual) functional test, and your tarball.